src/ralph/models/edx/converters/xapi/video.py

```diff
@@ -0,0 +1,178 @@
+"""Video event xAPI Converter"""
+
+from ralph.models.converter import ConversionItem
+from ralph.models.edx.video.statements import (
+    UILoadVideo,
+    UIPauseVideo,
+    UIPlayVideo,
+    UISeekVideo,
+    UISpeedChangeVideo,
+    UIStopVideo,
+)
+from ralph.models.xapi.constants import LANG_EN_US_DISPLAY
+from ralph.models.xapi.video.constants import (
+    VIDEO_EXTENSION_SPEED,
+    VIDEO_EXTENSION_TIME,
+    VIDEO_EXTENSION_TIME_FROM,
+    VIDEO_EXTENSION_TIME_TO,
+)
+from ralph.models.xapi.video.statements import (
+    VideoInitialized,
+    VideoInteracted,
+    VideoPaused,
+    VideoPlayed,
+    VideoSeeked,
+    VideoTerminated,
+)
+
+from .base import BaseXapiConverter
+
+
+class UILoadVideoToVideoInitialized(BaseXapiConverter):
+    """Converts a common edX `load_video` event to xAPI."""
+
+    __src__ = UILoadVideo
+    __dest__ = VideoInitialized
+
+    def _get_conversion_items(self):
+        """Returns a set of ConversionItems used for conversion."""
+
+        conversion_items = super()._get_conversion_items()
+        return conversion_items.union(
+            {
+                ConversionItem(
+                    "object__name",
+                    "event__id",
+                    lambda id: {LANG_EN_US_DISPLAY.__args__[0]: id},
+                )
+            },
+        )
+
+
+class UIPlayVideoToVideoPlayed(BaseXapiConverter):
+    """Converts a common edX `play_video` event to xAPI."""
+
+    __src__ = UIPlayVideo
+    __dest__ = VideoPlayed
+
+    def _get_conversion_items(self):
+        """Returns a set of ConversionItems used for conversion."""
+
+        conversion_items = super()._get_conversion_items()
+        return conversion_items.union(
+            {
+                ConversionItem(
+                    "object__name",
+                    "event__id",
+                    lambda id: {LANG_EN_US_DISPLAY.__args__[0]: id},
+                ),
+                ConversionItem(
+                    "result__extensions__" + VIDEO_EXTENSION_TIME,
+                    "event__currentTime",
+                ),
+            },
+        )
+
+
+class UIPauseVideoToVideoPaused(BaseXapiConverter):
+    """Converts a common edX `pause_video` event to xAPI."""
+
+    __src__ = UIPauseVideo
+    __dest__ = VideoPaused
+
+    def _get_conversion_items(self):
+        """Returns a set of ConversionItems used for conversion."""
+
+        conversion_items = super()._get_conversion_items()
+        return conversion_items.union(
+            {
+                ConversionItem(
+                    "object__name",
+                    "event__id",
+                    lambda id: {LANG_EN_US_DISPLAY.__args__[0]: id},
+                ),
```

Comment:
How can we define the xAPI `object__id` from the edX video_paused event?
Could we use the edX `event__id` for it instead? WDYT?

Reply:
Sorry, I forgot that the object ID should be an IRI.  However, one option might be to concatenate the edX `page` field with the `event__id`  to obtain an IRI?